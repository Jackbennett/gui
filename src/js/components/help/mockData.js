export const helpProps = {
  docsVersion: '',
  findLocalIpAddress: jest.fn,
  isHosted: false,
  isEnterprise: false,
  menderDebPackageVersion: 'master',
  menderVersion: 'master',
  menderArtifactVersion: 'master',
  org: {
    tenant_token: ''
  }
};
