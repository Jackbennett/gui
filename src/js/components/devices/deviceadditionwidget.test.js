import React from 'react';
import { act, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import DeviceAdditionWidget from './deviceadditionwidget';
import { undefineds } from '../../../../tests/mockData';

describe('DeviceAdditionWidget Component', () => {
  it('renders correctly', async () => {
    const { baseElement } = render(<DeviceAdditionWidget docsVersion="" onConnectClick={jest.fn} />);
    const view = baseElement.firstChild.firstChild;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });

  it('works as intended', async () => {
    const clickMock = jest.fn();
    render(<DeviceAdditionWidget docsVersion="" onConnectClick={clickMock} />);
    act(() => userEvent.click(screen.getByRole('button', { name: /connect a new device/i })));
    expect(clickMock).toHaveBeenCalled();
  });
});
