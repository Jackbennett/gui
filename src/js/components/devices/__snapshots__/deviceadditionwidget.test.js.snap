// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceAdditionWidget Component renders correctly 1`] = `
<div
  class="flexbox column center-aligned padding-small device-addition-widget"
>
  <div
    class="MuiButtonGroup-root muted"
    role="group"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined"
      style="text-transform: none;"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Connect a new device
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
      style="text-transform: none;"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
  <a
    class="flexbox centered"
    href="https://docs.mender.io/client-installation/overview"
    rel="noopener noreferrer"
    style="margin-top: 8px;"
    target="_blank"
  >
    Learn how to connect devices
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root"
      focusable="false"
      style="font-size: 10pt; margin-left: 8px; margin-bottom: -2px;"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
      />
    </svg>
  </a>
</div>
`;
